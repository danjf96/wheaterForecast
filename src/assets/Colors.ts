export default {
    principal: '#0086ff',
    secondary: 'white',

    text: 'black',
    textSecondary: 'white',

    title: 'black',

    backgroundModal: '#353535',
}