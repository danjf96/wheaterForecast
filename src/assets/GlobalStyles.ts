import { StyleSheet } from "react-native";
import Colors from "./Colors";

export const LOGO = require('../assets/images/logo.png')

export const GlobalStyle = StyleSheet.create({
    line: {
        width: '100%', 
        marginVertical: 25, 
        borderWidth: 0.3,
        borderColor: Colors.principal,
        opacity: 0.2
    }
})
