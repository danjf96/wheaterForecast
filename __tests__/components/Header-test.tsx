
/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import Header from '../../src/components/header'
 // Note: test renderer must be required after react-native.
 import { fireEvent, render } from 'react-native-testing-library';
import { RELOAD } from '../../src/assets/GlobalStyles';
 
 const fakeValue = {
    text: 'teste', 
    showIcon: true, 
    onPress: jest.fn(), 
    icon: RELOAD
}

it('renders correctly', async () => {
    const screen = render(<Header {...fakeValue}/>)
    expect(screen.toJSON()).toMatchSnapshot()
});

describe('more tests', () => {

    it('values', async () => {
        const { getAllByText, getByTestId } = render(<Header {...fakeValue}/>)

        expect(getAllByText(/teste/)).toBeTruthy()

        const iconReload = getByTestId('reload')
        fireEvent(iconReload, 'press')
        expect(fakeValue.onPress).toBeCalled()

    });
})
 