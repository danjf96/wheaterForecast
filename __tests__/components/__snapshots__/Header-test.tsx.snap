// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    teste
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="reload"
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/reload.png",
        }
      }
      style={
        Object {
          "height": 48,
          "marginLeft": 15,
          "resizeMode": "contain",
          "width": 48,
        }
      }
    />
  </View>
</View>
`;
